File: Practicas/Yahir-Hermosillo/Parcial3/Practica6/MagicPerson.java
public class MagicPerson {
    private String name;
    private int age;
    private String house;

    public MagicPerson(String name, int age){
        this.name = name;
        this.age = age;
    }

    public void introduce() {
        System.out.println("Hola, soy " + name + ", tengo " + age + " años" + 
                          (house != null ? " y soy de " + house : "") + ".");
    }
           
        
    public void castSpell() {
        System.out.println(name + " lanza un hechizo genérico.");
    }

    public String getName() {
        return name;
    }

    public void setHouse(String house) {
        this.house = house;
    }
}

File: Practicas/Marco-Oregel/Practica6/MagicPerson.py
class MagicPerson:

    def __init__(self,name,age, house = None):
        self.name = name
        self.age = age
        self.house = house

    def castSpell(self):
        return "Abracadabra!"
    
    def introduce(self):
        chosenHouse = f" y soy de la casa {self.house}" if self.house else " "
        print(f"Hola, mi nombre es {self.name}, tengo {self.age} años{chosenHouse}.")